clang/lib/CIR/CodeGen/CIRGenOpenACCClause.h

```diff
@@ -82,6 +83,17 @@ class OpenACCClauseCIREmitter final
     return conversionOp.getResult(0);
   }
 
+  mlir::Value createConstantInt(mlir::Location loc, unsigned width,
+                                int64_t value) {
+    mlir::IntegerType ty = mlir::IntegerType::get(
+        &cgf.getMLIRContext(), width,
+        mlir::IntegerType::SignednessSemantics::Signless);
+    auto constOp = builder.create<mlir::arith::ConstantOp>(
+        loc, ty, builder.getIntegerAttr(ty, value));
```

Comment:
Its a little unfortunate that this doesn't reuse the constant.  Unless I'm missing a good way for these to auto-canonicalize?  or do we just expect the simplify passes will take care of that?

Reply:
The way to get the generic canonicalizer pass to fold constants is to implement `::fold` methods (and declare ops with `hasFolder = 1`) and then run the canonicalizer pass. Alternatively you can add the op to CIRCanonicalizer as well, which will call into the same apply greedy rewrite stuff.

Reply:
As these are mlir types, do you think it makes sense to go modify the MLIR types for this canonicalization?  Really it doesn't affect anything, but it just seemed like we were creating more than we should have, and I wanted to make sure that wasn't problematic.